### PR TITLE
Replace osp::BitVector_t with lgrn::IdSetStl

### PR DESCRIPTION
**Pull request by Lamakaio**
This pull request replaces osp::BitVector_t with lgrn::IdSetStl as outlined in issue #282. It depends on #281 and #284 at the moment. 

All the functional refactoring should be done, but for some reason both this branch and #281 fail to compile on my (admittedly janky) hardware, and I still need to do a pass to check formatting and variable names.

### CODE REVIEW

**Comment by Capital-Asterisk** [comment]
Delightfully surprised to see this addressed rather quickly, thanks!

May I ask what the build issue might be? If it's related to building SDL2, then there exists the `-DOSP_USE_SYSTEM_SDL=ON` option that can be enabled to use SDL2 through a system package manager or other means.

**Comment by Lamakaio** [comment]
I solved the build issue. Somehow my computer tends to crash during builds (not only this project) and it corrupted some compilation artifacts or something, entirely on my end anyway. 

However, I am now having some issues on assertion fail due to insufficient size for the IdSets (trying to insert bit 64 on set of size 64). 

In some places in the code, sets are resized like that  (in testapp/sessions/shapes.cpp)
```cpp
rPhys.m_hasColliders.resize(rBasic.m_activeIds.vec().capacity());
```

The resize function looks like that (in id_set_htl.hpp)
```cpp
void resize(std::size_t n)
    {
        vec().resize(lgrn::div_ceil(n, Base_t::bitview().int_bitsize()), 0);
    }
```

To me, it looks like we're using the capacity of an uint64 vector as the number of individual bits in the IdSet, and that is why it is too small.

However, I cannot figure out why it would work before if that was the case, as my changes should not have changed that behavior.

Anyway, I get assertion fails on insert on the sets that are resized this way (after the refactor).

**Comment by jonesmz** [comment]
> Somehow my computer tends to crash during builds (not only this project) and it corrupted some compilation artifacts or something, entirely on my end anyway.

Try limiting parallelism in the build.

`cmake --build path/to/build --parallel numberParallelJobs``

**Comment by Lamakaio** [comment]
Thank you, that makes a lot of sense !

I fixed it and rebased the PR.

File: src/testapp/enginetest.cpp [hunk 1]
```diff
@@ -133,8 +133,8 @@ entt::any setup_scene(osp::Resources& rResources, osp::PkgId const pkg)
 
     // Resize some containers to fit all existing entities
     std::size_t const maxEnts = rScene.m_activeIds.vec().capacity();
-    rScene.m_matPhong.ints()    .resize(maxEnts);
```
Comment (Capital-Asterisk) [inline src/testapp/enginetest.cpp:136]: looks like this was oversized by 64x previously

**Review by Capital-Asterisk — APPROVED** [review]
looks good!

**Pull request merged**